activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -125,6 +125,21 @@ def schema_creation # :nodoc:
         PostgreSQL::SchemaCreation.new self
       end
 
+      def column_spec_for_primary_key(column)
+        spec = {}
+        if column.serial?
+          return unless column.sql_type == 'bigint'
+          spec[:id] = ':bigserial'
+        elsif column.type == :uuid
+          spec[:id] = ':uuid'
+          spec[:default] = column.default_function.inspect
+        else
+          spec[:id] = column.type.inspect
+          spec.merge!(prepare_column_options(column).delete_if { |key, _| [:name, :type, :null].include?(key) })
```

Comment:
Same thing here: `prepare_column_options` in 4.2 requires two arguments: `column` and `types`.